test/alternator/test_item.py

```diff
@@ -145,6 +145,23 @@ def test_item_operations_nonexistent_table(dynamodb):
         dynamodb.meta.client.put_item(TableName='non_existent_table',
             Item={'a':{'S':'b'}})
 
+# DynamoDB documentation specifies that table names must be 3-255 characters,
+# and match the regex [a-zA-Z0-9._-]+. If an item operation like PutItem
+# is given a TableName not matching these rules, it would make sense to treat
+# it as just another case of a non-existent table (ResourceNotFoundException),
+# but it turns out that DynamoDB returns a ValidationError in that case.
+# Saying for example that "Value 'non!existent!table' at 'tableName' failed
+# to satisfy constraint: Member must satisfy regular expression pattern:
+# [a-zA-Z0-9_.-]+".".
+# This test suggests that issue #12538 should not be done.
```

Comment:
it could be done in reverse order (like on catch from lower layer), although after getting rid of regex it's likely not worth it

Reply:
You're right. We can first check for invalid name **after** we discover the table doesn't exist. I think it's a good idea, even if the benefit is small now.